app/models/level.rb

```diff
@@ -0,0 +1,21 @@
+class Level < ActiveRecord::Base
+  belongs_to :course
+  validates :num, presence: true, numericality: true, uniqueness: { scope: :course_id }
+  validates :color, presence: true, uniqueness: { scope: :course_id }
+  validates :title, presence: true
+  validates :level_description, presence: true
+
+  # level_description is an array that is getting saved as a string
+  # this parses it back as an array
+  def description
```

Comment:
there's probably a more elegant way to do this with [serialize](http://apidock.com/rails/ActiveRecord/Base/serialize/class)